airbyte-server/src/main/java/io/airbyte/server/handlers/WebBackendConnectionsHandler.java

```diff
@@ -173,6 +175,25 @@ private WebBackendConnectionRead buildWebBackendConnectionRead(final ConnectionR
       webBackendConnectionRead.setLatestSyncJobStatus(job.getStatus());
     });
 
+    SchemaChange schemaChange = SchemaChange.NO_CHANGE;
+
+    if (currentSourceCatalogId.isPresent()) {
+      final Optional<ActorCatalogFetchEvent> mostRecentFetchEvent =
+          configRepository.getMostRecentActorCatalogFetchEventForSource(connectionRead.getSourceId());
+
+      if (mostRecentFetchEvent.isPresent()) {
+        if (!mostRecentFetchEvent.get().getActorCatalogId().equals(currentSourceCatalogId.get())) {
+          if (connectionRead.getBreakingChange()) {
```

Comment:
Trying to get a bit more context to understand what's happening here, when do we check if the change is breaking?

Reply:
this is constructing the WebBackendConnectionRead object to return to the front end, and we need to populate the schemaChange field with either "no_change", "non_breaking" or "breaking"

Reply:
It looks like we already checked if it was a `breakingChange` in `connectionRead`, why do we need this extra logic about checking the `currentSourceCatalogId` and the `mostRecentFetchEvent`?

Reply:
i commented outside of the thread: @gosusnp the part we're reading from the db is whether the connection has a breaking change. if that value is false, there still may be a non-breaking schema change and we get that by checking the actor catalog fetch events -- i could change the order of this logic so for cases where there are breaking changes we don't have to read the actor catalog fetch events

Reply:
I just updated so that we first check whether the connectionread object has breakingChange=true, and then after that we do the ActorCatalogFetchEvent check

Reply:
I find that a lot clearer, thanks!